apps/wallet/src/ui/app/pages/home/transfer-coin/SendTokenForm.tsx

```diff
@@ -44,6 +49,55 @@ export type SendTokenFormProps = {
     initialTo: string;
 };
 
+function GasBudgetEstimation({
+    coinDecimals,
+    coins,
+}: {
+    coinDecimals: number;
+    coins: CoinStruct[];
+}) {
+    const activeAddress = useActiveAddress();
+    const { values, setFieldValue } = useFormikContext<FormValues>();
+
+    const transaction = useMemo(() => {
+        if (!values.amount || !values.to || !coins) return null;
+
+        return createTokenTransferTransaction({
+            to: values.to,
+            amount: values.amount,
+            coinType: SUI_TYPE_ARG,
+            coinDecimals,
+            isPayAllSui: values.isPayAllSui,
+            coins,
+        });
+    }, [coinDecimals, coins, values.amount, values.isPayAllSui, values.to]);
+
+    const { data: gasBudget } = useTransactionGasBudget(
+        activeAddress,
+        transaction
+    );
+
+    // gasBudgetEstimation should change when the amount above changes
+    useEffect(() => {
+        setFieldValue('gasBudgetEst', gasBudget, true);
+    }, [gasBudget, setFieldValue, values.amount]);
+
+    if (!gasBudget) return null;
```

Comment:
We could always show this and instead of showing an amount show something to indicate it's loading or - or something